### PR TITLE
docs: update installation documentation

### LINKED ISSUE #1995: docs: update install procedures to use yarn
We recently migrated our testing framework to use yarn (#1966) as the default package manager in our testing builds.  This was because of issues found using npm v5 as well as a desire to have strictly reproducible builds using lock files.

We should update our documentation such as README and INSTALL files to reflect a change to yarn as the default package manager.

### PR DESCRIPTION
**Pull request by jniles**
This commit updates the installation documentation to clarify procedures and remove old cruft.  I've additionally changed references to `npm` to `yarn` to reflect what is being used in Travis.

Closes #1995.  Contributes to #2581.

Note the massive numbers of changes is due to me bumping yarn's lockfile.

### CODE REVIEW

File: docs/INSTALL.md [hunk 1]
```diff
@@ -3,69 +3,106 @@ Bhima Installation Guide
...
-
-Alternative, you might use the `build-database.sh` script, customized with your environmental variables as shown below:
+_NOTE: BHIMA runs in `sql_mode='STRICT_ALL_TABLES'`.  While it is not necessary
+to have this set to build the database, the test will not pass unless the
+correct SQL_MODE is set._
+
+The database structure is contained in the `server/models/*.sql` files.  You can
+execute these one by one in the order below, or simply run `yarn build:db`.
+ 1. `server/models/schema.sql`
+ 2. `server/models/.sql`
```
Comment (mbayopanda) [inline docs/INSTALL.md]: Is this file `server/models/.sql` exists ?

**Review by mbayopanda — COMMENTED** [review]
👍 LGTM

There is just one question about a `server/models/.sql` file

File: docs/INSTALL.md (hunk 1 above)
Comment (jniles) [reply docs/INSTALL.md]: Ha!  Good catch :)

**Comment by jniles** [comment]
@mbayopanda changes landed.

**Pull request merged**